mimir/client.go

```diff
@@ -66,13 +72,23 @@ func (t *BasicAuthTransport) RoundTrip(req *http.Request) (*http.Response, error
 
 // Ready checks if mimir is ready to serve traffic.
 func (c *Client) Ready(ctx context.Context) error {
-	path := c.address.JoinPath("/ready")
+	path := c.address.JoinPath("/")
+
+	if c.deploymentMode == "distributed" {
+		path = c.address.JoinPath("/api/v1/status/buildinfo")
+	} else {
+		path = c.address.JoinPath("/ready")
+	}
 
 	req, err := http.NewRequestWithContext(ctx, http.MethodGet, path.String(), nil)
 	if err != nil {
 		return err
 	}
 
+	if c.orgID != "" {
+		req.Header.Set("X-Scope-OrgID", c.orgID)
```

Comment:
Let's make `X-Scope-OrgID` a constant given that we reference it multiple times.